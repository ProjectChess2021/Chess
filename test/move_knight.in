game human human
move b1 c3
move b8 a6
move c3 a4
move g8 h6
move a4 c5
move h6 g4
move c5 a6
move g4 h2
move a6 b4