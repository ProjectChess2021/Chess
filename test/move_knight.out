8 rnbqkbnr      MoveHistory
7 pppppppp
6  _ _ _ _
5 _ _ _ _
4  _ _ _ _
3 _ _ _ _
2 PPPPPPPP
1 RNBQKBNR

  abcdefgh
8 rnbqkbnr                                                         MoveHistory (Newest at top)
7 pppppppp                                                         White: b1 -> c3
6  _ _ _ _                                                         
5 _ _ _ _                                                          
4  _ _ _ _                                                         
3 _ N _ _                                                          
2 PPPPPPPP                                                         
1 R BQKBNR                                                            

  abcdefgh
8 r bqkbnr      MoveHistory
7 pppppppp      White:knight b1 -> c3
6 n_ _ _ _      Black:knight b8 -> a6
5 _ _ _ _
4  _ _ _ _
3 _ N _ _
2 PPPPPPPP
1 R BQKBNR

  abcdefgh
8 r bqkbnr      MoveHistory
7 pppppppp      White:knight b1 -> c3
6 n_ _ _ _      Black:knight b8 -> a6
5 _ _ _ _       White:knight c3 -> a4
4 N_ _ _ _
3 _ _ _ _
2 PPPPPPPP
1 R BQKBNR

  abcdefgh
8 r bqkb r      MoveHistory
7 pppppppp      White:knight b1 -> c3
6 n_ _ _ n      Black:knight b8 -> a6
5 _ _ _ _       White:knight c3 -> a4
4 N_ _ _ _      Black:knight g8 -> h6
3 _ _ _ _
2 PPPPPPPP
1 R BQKBNR

  abcdefgh
8 r bqkb r      MoveHistory
7 pppppppp      White:knight b1 -> c3
6 n_ _ _ n      Black:knight b8 -> a6
5 _ N _ _       White:knight c3 -> a4
4  _ _ _ _      Black:knight g8 -> h6
3 _ _ _ _       White:knight a4 -> c5
2 PPPPPPPP
1 R BQKBNR

  abcdefgh
8 r bqkb r      MoveHistory
7 pppppppp      White:knight c3 -> a4
6 N_ _ _ _      Black:knight g8 -> h6
5 _ _ _ _       White:knight a4 -> c5
4  _ _ _n_      Black:knight h6 -> g4
3 _ _ _ _       White:knight c5 -> a6 (capture a knight)
2 PPPPPPPP
1 R BQKBNR

  abcdefgh
8 r bqkb r      MoveHistory
7 pppppppp      Black:knight g8 -> h6
6 N_ _ _ _      White:knight a4 -> c5
5 _ _ _ _       Black:knight h6 -> g4
4  _ _ _ _      White:knight c5 -> a6 (capture a knight)
3 _ _ _ _       Black:knight g4 -> h2 (capture a pawn)
2 PPPPPPPn
1 R BQKBNR

  abcdefgh
8 r bqkb r      MoveHistory
7 pppppppp      White:knight a4 -> c5
6  _ _ _ _      Black:knight h6 -> g4
5 _ _ _ _       White:knight c5 -> a6 (capture a knight)
4  N _ _ _      Black:knight g4 -> h2 (capture a pawn)
3 _ _ _ _       White:knight a6 -> b4
2 PPPPPPPn
1 R BQKBNR

  abcdefgh
